### PR TITLE
Ladybird: Close tab clicking wheel of the mouse

### PR DESCRIPTION
**Pull request by guerinoni**

### CODE REVIEW

File: Ladybird/BrowserWindow.cpp [hunk 1]
```diff
@@ -504,3 +506,19 @@ void BrowserWindow::copy_selected_text()
         clipboard->setText(qstring_from_ak_deprecated_string(text));
     }
 }
+
+bool BrowserWindow::eventFilter(QObject* obj, QEvent* event)
```
Comment (elcuco) [inline Ladybird/BrowserWindow.cpp:510]: Doesn't this affect "drag"? Why not filtering the middle button click instead?

File: Ladybird/BrowserWindow.cpp [hunk 2]
```diff
@@ -30,6 +31,7 @@ BrowserWindow::BrowserWindow(Browser::CookieJar& cookie_jar, StringView webdrive
     : m_cookie_jar(cookie_jar)
     , m_webdriver_content_ipc_path(webdriver_content_ipc_path)
 {
+    qApp->installEventFilter(this);
```
Comment (elcuco) [inline Ladybird/BrowserWindow.cpp:34]: ... why not installing the filter at the window level...?

File: Ladybird/BrowserWindow.cpp (hunk 1 above)
Comment (guerinoni) [reply Ladybird/BrowserWindow.cpp:510]: The drag is working properly, what do you mean for filtering for middle button?

File: Ladybird/BrowserWindow.cpp (hunk 2 above)
Comment (guerinoni) [reply Ladybird/BrowserWindow.cpp:34]: At window level I don't have accetto to tab_widget

Comment (elcuco) [reply Ladybird/BrowserWindow.cpp:34]: Use this instead (this works in line 40 after the creation of `m_tabs_container`):
```
m_tabs_container->installEventFilter(this);
``` 

This is a less "abusive" filter.

File: Ladybird/BrowserWindow.cpp (hunk 1 above)
Comment (elcuco) [reply Ladybird/BrowserWindow.cpp:510]: OK, no problem here, my "Qt" is a little rusty :)

File: Ladybird/BrowserWindow.cpp (hunk 2 above)
Comment (guerinoni) [reply Ladybird/BrowserWindow.cpp]: Oh thank you :)

**Review by elcuco — APPROVED** [review]

File: Ladybird/BrowserWindow.cpp [hunk 3]
```diff
@@ -504,3 +506,19 @@ void BrowserWindow::copy_selected_text()
         clipboard->setText(qstring_from_ak_deprecated_string(text));
     }
 }
+
+bool BrowserWindow::eventFilter(QObject* obj, QEvent* event)
+{
+    if (event->type() == QEvent::MouseButtonRelease) {
```
Comment (konradekk) [inline Ladybird/BrowserWindow.cpp:512]: is it possible to middle click outside the tab and release the button on it to close it? 🤔

itʼd be good to have proper clicks only, i.e. pressing middle button on the tab and releasing it there…

Comment (guerinoni) [reply Ladybird/BrowserWindow.cpp:512]: It works only if you press and release staying on the tab :)

Comment (konradekk) [reply Ladybird/BrowserWindow.cpp:512]: okay; was not sure! 😅

**Review by konradekk — APPROVED** [review]

**Review by AtkinsSJ — APPROVED** [review]

**Pull request merged**